troposphere/sqs.py

```diff
@@ -34,6 +34,12 @@ class Queue(AWSObject):
         'VisibilityTimeout': (integer, False),
     }
 
+    def validate(self):
+        if self.properties.get('FifoQueue'):
+            if not self.properties.get('QueueName', '').endswith('.fifo'):
```

Comment:
Strangely, that's the first thing I thought too when we started using this internally.  But after testing, it turns out that if you specify FifoQueue, you *have* to specify a QueueName, and it has to end with `.fifo`.  It's annoying, but CF isn't smart enough to understand it has to add `.fifo`, which isn't all that surprising after all this time of messing w/ CF :)